packages/currency/src/Balance.ts

```diff
@@ -69,19 +77,29 @@ export default class Balance<T extends BaseCurrencyType> {
     const groupSeparator = options?.groupSeparator || ','
     const showTicker = options?.showTicker === undefined ? true : options.showTicker
     const format = { decimalSeparator, groupSeparator, groupSize: 3 }
+    const roundingMode = options?.roundingMode || BigNumber.ROUND_DOWN
+    const decimalPlacesToDisplay = getVal(maxDecimalPlaces, this.type.format?.decimalPlaces)
+    const keepTrailingZeroes = getVal(
+      options?.showTrailingZeroes,
+      this.type.format?.showTrailingZeroes,
+    )
 
     let numberString = ''
-    if (maxDecimalPlaces !== undefined && maxDecimalPlaces !== null) {
-      numberString = this.bigBalance.toFormat(
-        maxDecimalPlaces,
-        options?.roundingMode || BigNumber.ROUND_DOWN,
-        format,
-      )
+    if (decimalPlacesToDisplay !== undefined && decimalPlacesToDisplay !== null) {
+      let decimalPlaces = decimalPlacesToDisplay
+
+      if (!keepTrailingZeroes) {
+        decimalPlaces = Math.min(
+          decimalPlacesToDisplay,
+          this.bigBalance.decimalPlaces(decimalPlacesToDisplay, roundingMode).decimalPlaces(),
+        )
+      }
+      numberString = this.bigBalance.toFormat(decimalPlaces, roundingMode, format)
     } else {
       numberString = this.bigBalance.toFormat(format)
     }
-    // if it's an integer, just show the integer
-    if (parseInt(numberString.split(decimalSeparator)[1], 10) === 0) {
+    // if it's an integer and keepTrailingZeroes is false, just show the integer
+    if (!keepTrailingZeroes && parseInt(numberString.split(decimalSeparator)[1], 10) === 0) {
```

Comment:
This could be better in case there is no decimal for some reason
```suggestion
    if (!keepTrailingZeroes) {
```

Reply:
Not sure I follow... I actually think that whole if block can be removed now.

Reply:
That too. I was just saying we don't need to check if there is a trailing zero because the split would work anyway